### PR TITLE
RUMM-3374 fix: Manual trace injection APIs are not available in `DatadogTrace`

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

🐞 This PR fixes the problem of `*HTTPHeaderWriter` APIs being not available after importing `DatadogTrace` in `2.0.0`.

It was a miss during `1.x` → `2.x` APIs migration. As a workaround, users had to import `DatadogInternal` directly into their code. These APIs should be available with only `import DatadogTrace`.

### How?

The actual fix are following exports in `DatadogTrace` module:
```swift
@_exported import class DatadogInternal.HTTPHeadersWriter
@_exported import class DatadogInternal.B3HTTPHeadersWriter
@_exported import class DatadogInternal.W3CHTTPHeadersWriter
```

Extra work done:
* For coherence with other APIs, I aligned naming `samplingRate` → `sampleRate`.
* The `OTelHTTPHeaders*` APIs are now deprecated and renamed to `B3HTTPHeaders*`, see https://github.com/DataDog/dd-sdk-ios/pull/1415#discussion_r1296035255
* I fixed few issues and improved the general sense of API comments.
* Last, I added debug screen to `Example` project to manually test if `import DatadogTrace` will make writers work (unit tests are already in place):

<p align="center">
Screenshot 2023-08-16 at 13 36 19
</p>

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [x] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: DatadogInternal/Sources/NetworkInstrumentation/OpenTelemetry/OTelHTTPHeadersWriter.swift [hunk 1]
```diff
@@ -6,41 +6,45 @@
...
+///     Tracer.shared().inject(spanContext: span.context, writer: writer)
 ///
 ///     writer.traceHeaderFields.forEach { (field, value) in
 ///         request.setValue(value, forHTTPHeaderField: field)
 ///     }
 ///
 ///     // call span.finish() when the request completes
 ///
-///
 public class OTelHTTPHeadersWriter: TracePropagationHeadersWriter {
```
Comment (ncreated) [inline DatadogInternal/Sources/NetworkInstrumentation/OpenTelemetry/OTelHTTPHeadersWriter.swift:26]: 💡 Even with these changes, I'm concerned about the `OTelHTTPHeadersWriter` name. OTel doesn't dictate this or that headers format and using "OTel" definitely does not imply `B3-*` headers. It should be `B3HTTPHeadersWriter` IMO (consistent with `W3CHTTPHeadersWriter` and bare `HTTPHeadersWriter` which uses `x-datadog-*` headers).

I'd be up for deprecating this symbol and renaming it accordingly. WDYT @ganeshnj @maciejburda @maxep ?

**Review by maciejburda — DISMISSED** [review]
Great stuff! I like renaming proposal

File: DatadogInternal/Sources/NetworkInstrumentation/OpenTelemetry/OTelHTTPHeadersWriter.swift (hunk 1 above)
Comment (maxep) [reply DatadogInternal/Sources/NetworkInstrumentation/OpenTelemetry/OTelHTTPHeadersWriter.swift:26]: [Agreed](https://github.com/DataDog/dd-sdk-ios/pull/1057#pullrequestreview-1187895880) :) `B3*` is technically more correct 👍

**Review by maxep — DISMISSED** [review]
Nice! Just for clarification: why only in `DatadogTrace` and not in `DatadogRUM`?

File: DatadogInternal/Sources/NetworkInstrumentation/OpenTelemetry/OTelHTTPHeadersWriter.swift (hunk 1 above)
Comment (ncreated) [reply DatadogInternal/Sources/NetworkInstrumentation/OpenTelemetry/OTelHTTPHeadersWriter.swift]: Renamed `OTelHTTPHeaders*` to `B3HTTPHeaders*` with deprecating the prior. See: https://github.com/DataDog/dd-sdk-ios/pull/1415/commits/4a3819c1f79fcb83f6dbaa12aee1d394906848cb

**Review by maxep — APPROVED** [review]

**Review by maxep — APPROVED** [review]

**Pull request merged**